.pre-commit-config.yaml

```diff
@@ -0,0 +1,56 @@
+# pre-commit is a tool that you run locally
+# to perform a predefined set of tasks manually and/or
+# automatically before git commits are made. 
+# Here we are using pre-commit with the precommit.ci bot to implement 
+# code fixes automagically in pr's. You will still want to install pre-commit 
+# to run locally
+# Config reference: https://pre-commit.com/#pre-commit-configyaml---top-level
+# To run on a pr, use: pre-commit.ci run
+# Common tasks
+#
+# - Run on all files:   pre-commit run --all-files
+# - Register git hooks: pre-commit install --install-hooks
+
+ci:
+    autofix_prs: false
+    skip: [black]
+    #autofix_commit_msg: |
+        #'[pre-commit.ci 🤖] Apply code format tools to PR'
+    # Update hook versions every quarter (so we don't get hit with weekly update pr's)
+    autoupdate_schedule: quarterly
+
+repos:
+  - repo: https://github.com/PyCQA/isort
+    rev: 5.11.4
+    hooks:
+      - id: isort
+        files: \.py$
+
+  # Misc commit checks
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v4.4.0
+    # ref: https://github.com/pre-commit/pre-commit-hooks#hooks-available
+    hooks:
+      # Autoformat: Makes sure files end in a newline and only a newline.
+      - id: end-of-file-fixer
+
+      # Lint: Check for files with names that would conflict on a
+      # case-insensitive filesystem like MacOS HFS+ or Windows FAT.
+      - id: check-case-conflict
+      - id: trailing-whitespace
+
+  # Linting: Python code (see the file .flake8)
+  - repo: https://github.com/PyCQA/flake8
+    rev: "6.0.0"
+    hooks:
+      - id: flake8
+
+# Black for auto code formatting
+repos:
+- repo: https://github.com/psf/black
+  rev: 22.12.0
+  hooks:
+  - id: black
+    language_version: python3.8
```

Comment:
Why is there a second `repos` list? Shouldn't black be a part of the first `repos` list? My IDE also cries about this construct being invalid YAML.

Reply:
great catch! I've fixed this. It was because i added and removed black and worked too quickly!! definitely a mistake.